libs/model/src/models/evmEventSource.ts

```diff
@@ -34,20 +34,26 @@ export default (
         chain_node_id: {
           type: dataTypes.INTEGER,
           allowNull: false,
+          unique: 'unique_event_source',
+        },
+        contract_address: {
+          type: dataTypes.STRING,
+          allowNull: false,
+          unique: 'unique_event_source',
+        },
+        event_signature: {
+          type: dataTypes.STRING,
+          allowNull: false,
+          unique: 'unique_event_source',
+        },
+        kind: {
+          type: dataTypes.STRING,
+          allowNull: false,
         },
-        contract_address: { type: dataTypes.STRING, allowNull: false },
-        event_signature: { type: dataTypes.STRING, allowNull: false },
-        kind: { type: dataTypes.STRING, allowNull: false },
       },
       {
         tableName: 'EvmEventSources',
         timestamps: false,
-        indexes: [
```

Comment:
We should leave this here and instead add a additional `name` option. Moving the unique to the attribute definition means we can't provide a custom configuration for the index other than a name.

Reply:
if we create an index, it won't match the migrations schema that is based on a CONSTRAINT:

```sql
CREATE TABLE IF NOT EXISTS public."EvmEventSources"
(
    id integer NOT NULL DEFAULT nextval('"EvmEventSources_id_seq"'::regclass),
    chain_node_id integer NOT NULL,
    contract_address character varying(255) COLLATE pg_catalog."default" NOT NULL,
    event_signature character varying(255) COLLATE pg_catalog."default" NOT NULL,
    kind character varying(255) COLLATE pg_catalog."default" NOT NULL,
    CONSTRAINT "EvmEventSources_pkey" PRIMARY KEY (id),
==>  CONSTRAINT unique_event_source UNIQUE (chain_node_id, contract_address, event_signature),
    CONSTRAINT "EvmEventSources_chain_node_id_fkey" FOREIGN KEY (chain_node_id)
        REFERENCES public."ChainNodes" (id) MATCH SIMPLE
        ON UPDATE NO ACTION
        ON DELETE NO ACTION
)
```

- we can add a migration to remove the constraint and create an index, if that's what we want
- both are probably using the same underlying index, so maybe not a big deal to keep it like it is in the PR to match the migrations

Reply:
Let's leave it as constraint. Though worth noting that we cannot provide a name for the constraint so this means we will not be able to match constraint names in the future.